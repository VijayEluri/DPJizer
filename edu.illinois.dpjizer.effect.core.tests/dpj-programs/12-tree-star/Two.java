/**
 * This file is licensed under the University of Illinois/NCSA Open Source License. See LICENSE.TXT for details.
 */
package two;

/**
 * 
 * @author Mohsen Vakilian
 * @author Danny Dig
 *
 */
class C<region X> {
	region L;
	region Q;
	C<X:L> left in X:L;

	C() {}

	void m1() {
		left.m2();
	}

	void m2() {
		new C<Q>().m1();
	}
}