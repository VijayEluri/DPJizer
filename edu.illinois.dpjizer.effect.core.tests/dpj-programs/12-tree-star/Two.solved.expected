two.C.C() pure
two.C.m1() reads Root : two.C.Q : two.C.L, X : two.C.L
two.C.m2() reads Root : two.C.Q : two.C.L
